src/stream/src/task/barrier_manager.rs

```diff
@@ -713,27 +721,30 @@ impl LocalBarrierWorker {
     /// to find the root cause.
     fn send_barrier(
         &mut self,
-        barrier: &Barrier,
-        to_send: HashSet<ActorId>,
-        to_collect: HashSet<ActorId>,
-        table_ids: HashSet<TableId>,
+        barrier: Barrier,
+        graph_infos: HashMap<u32, PartialGraphInfo>,
     ) -> StreamResult<()> {
+        let all_to_collect = || {
```

Comment:
nits: this can be a varaible with eager initialization instead of a closure given that `all_to_collect()` is used multiple times.